src/networking.c

```diff
@@ -1569,7 +1576,7 @@ void freeClient(client *c) {
 
     /* If a client is protected, yet we need to free it right now, make sure
      * to at least use asynchronous freeing. */
-    if (c->flags & CLIENT_PROTECTED) {
+    if ((c->flags & CLIENT_PROTECTED) || (c->flags & CLIENT_PROTECTED_RDB_CHANNEL)) {
```

Comment:
I don't see the client taken out of `server.replicas_waiting_psync` in `freeClient`?